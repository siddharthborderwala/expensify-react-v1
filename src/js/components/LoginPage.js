import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from './../actions/auth';

export const LoginPage = ({ startLogin }) => (
	<div className="box-layout">
		<div className="box-layout__box">
			<h1 className="box-layout__heading">Expensify App</h1>
			<p className="box-layout__tag-line">It's time to get your expenses under control</p>
			<button onClick={startLogin} className="btn btn-login">
				Login with Google
			</button>
		</div>
	</div>
);

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
