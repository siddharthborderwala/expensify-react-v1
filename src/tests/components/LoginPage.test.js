import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from './../../js/components/LoginPage';

test('should render LoginPage component correctly', () => {
	const wrapper = shallow(<LoginPage />);
	expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
	const startLogin = jest.fn();
	const wrapper = shallow(<LoginPage startLogout={startLogin} />);
	wrapper.find('button').simulate('click');
	expect(startLogin).toHaveBeenCalled();
});