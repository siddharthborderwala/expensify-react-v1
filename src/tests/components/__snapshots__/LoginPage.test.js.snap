// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage component correctly 1`] = `
<div>
  <h3>
    Login ot the App
  </h3>
  <button>
    <Link
      to="/dashboard"
    >
      Login
    </Link>
  </button>
</div>
`;
